### PR TITLE
Fix bug: colorize and count the unrecognized files

### PR DESCRIPTION
**Pull request by jaredyam**
### Description

Fix the bug that cannot properly colorize and count the unrecognized files.
Give a look at the change of the color of *.gem file (a unrecognized file) and the bottom few lines in the following screenshots.

### BEFORE

Screen Shot 2020-09-07 at 4 10 58 PM

### AFTER

Screen Shot 2020-09-07 at 4 11 59 PM

- Relevant Issues : (none)
- Relevant PRs : (none)
- Type of change :
  - [ ] New feature
  - [x] Bug fix for existing feature
  - [ ] Code quality improvement
  - [ ] Addition or Improvement of tests
  - [ ] Addition or Improvement of documentation

### CODE REVIEW

**Comment by avdv** [comment]
Hi, thanks for the PR!

Alas, the CI job failed: 

https://travis-ci.org/github/athityakumar/colorls/jobs/724841883

Make sure that `bundle exec rake` succeeds locally.

**Comment by avdv** [comment]
I'll take a deeper look this evening, but why are the markdown files no longer colored after your changes?

**Comment by jaredyam** [comment]
> I'll take a deeper look this evening, but why are the markdown files no longer colored after your changes?

Since I prefer to highlight the md files in my workflow, I have assigned this file type a specific color in my color scheme, which could be a new feature of this repo. According to the context of `core.rb`, I have improved my local commit fixing this bug with a more consistent code style.

 I'm also looking at the failure part, but it seems that it's irrelevant to my changes and also cannot pass the original repo.

**Comment by avdv** [comment]
> I'm also looking at the failure part, but it seems that it's irrelevant to my changes and the original repo also cannot pass the test.

WDYM? The CI job on the master branch passed the last couple of builds, including the latest commit.

However, looking into this I can see what the problem is (it's always a good idea to describe that in the commit message). The point is that a *file* is assigned a generic icon / glyph in `files.yaml`, so looking up `:file` in the corresponding hash always succeeds, but `:file` is also used as the fallback key if a file's extension is not found.

Your proposed fix should correct that but the logic is hard to follow. I am preparing a PR including a test.
